### PR TITLE
Adding date fields param to tableToMarkdown

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/41269

## Description
Adding date_fields param to tableToMarkdown to display dates in a human-readable format.

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -1774,7 +1774,8 @@ def create_clickable_url(url):
     return '[{}]({})'.format(url, url)
 
 
-def tableToMarkdown(name, t, headers=None, headerTransform=None, removeNull=False, metadata=None, url_keys=None):
+def tableToMarkdown(name, t, headers=None, headerTransform=None, removeNull=False, metadata=None, url_keys=None,
+                    date_fields=None):
```
Comment (dantavori) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:1778]: please add the arg description in the docstring

File: Packs/Base/ReleaseNotes/1_13_37.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### CommonServerPython
+- Added the *date_fields* param to tableToMarkdown to support displaying dates in a human-readable format.
```
Comment (dantavori) [inline Packs/Base/ReleaseNotes/1_13_37.md:4]:
```suggestion
- Added the *date_fields* argument to the *tableToMarkdown* function to support displaying dates in a human-readable format.
```

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 3]
```diff
@@ -1865,8 +1866,21 @@ def headerTransform(s): return stringEscapeMD(s, True, True)  # noqa
...
+
+            entry_copy = entry.copy()
+
+            if date_fields:
+                for field in date_fields:
+
+                    try:
+                        entry_copy[field] = datetime.fromtimestamp(int(entry_copy[field])/1000).strftime('%Y-%m-%d %H:%M:%S')
+
+                    except BaseException:
```
Comment (dantavori) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:1878]:
```suggestion
                    except Exception:
```
BaseException is also ok but a bit too broad, no reason to use it

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 4]
```diff
@@ -1865,8 +1866,21 @@ def headerTransform(s): return stringEscapeMD(s, True, True)  # noqa
...
-            vals = [stringEscapeMD((formatCell(entry.get(h, ''), False) if entry.get(h) is not None else ''),
+
+            entry_copy = entry.copy()
+
+            if date_fields:
+                for field in date_fields:
+
+                    try:
+                        entry_copy[field] = datetime.fromtimestamp(int(entry_copy[field])/1000).strftime('%Y-%m-%d %H:%M:%S')
+
```
Comment (dantavori) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:1877]:
```suggestion
            entry_copy = entry.copy()
            if date_fields:
                for field in date_fields:
                    try:
                        entry_copy[field] = datetime.fromtimestamp(int(entry_copy[field])/1000).strftime('%Y-%m-%d %H:%M:%S')
```
try to put spaces only when you want to separate code blocks that handle different things

**Review by dantavori — CHANGES_REQUESTED** [review]
very nice, see my (minor) notes

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py (hunk 1 above)
Comment (yaakovpraisler) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:1778]: done.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 5]
```diff
@@ -2135,8 +2149,9 @@ def isCommandAvailable(cmd):
     return False
 
 
-def epochToTimestamp(epoch):
-    return datetime.utcfromtimestamp(epoch / 1000.0).strftime("%Y-%m-%d %H:%M:%S")
+def epochToTimestamp(epoch, utc=True):
+    return datetime.utcfromtimestamp(epoch / 1000.0).strftime("%Y-%m-%d %H:%M:%S") if utc \
+        else datetime.fromtimestamp(epoch / 1000.0).strftime("%Y-%m-%d %H:%M:%S")
```
[the remark this answers is not in the document]
Comment (yaakovpraisler) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py]: done.

**Review by dantavori — APPROVED** [review]

**Pull request merged**